src/main/java/pages/AllProductsPage.java

```diff
@@ -7,48 +7,77 @@
+    private By allProductElements = By.xpath("//div[@class=\"col-sm-4\"]//div[@class=\"productinfo text-center\"]//p");
+    private By AddToCartButtons = By.xpath("//div[@class=\"productinfo text-center\"]//a[@class=\"btn btn-default add-to-cart\"]");
+    private By continueShopping = By.xpath("//button[@class='btn btn-success close-modal btn-block']");
+    private By viewCartButton = By.xpath("//a[@href=\"/view_cart\"]//u");
 
     public AllProductsPage(WebDriver driver) {
         super(driver);
     }
-    public ProductPage searchForProduct(String productName){
-        TypeOnFields(searchProductField,productName);
-        ElementClick(searchButton);
+
+    public ProductPage searchForProduct(String productName) {
+        typeOnFields(searchProductField, productName);
+        elementClick(searchButton);
         return new ProductPage(driver);
 
     }
-    public String getAllProductText(){
-       return GetElementText(allProductText);
+
+    public String getAllProductText() {
+        return getElementText(allProductText);
     }
-public List<WebElement> get_all_products(){
-    return get_list_of_elements(allProductElements);
-}
-public List<String> getProductsName(){
-        List<String>allProducts=new ArrayList<>();
-        for(int i=0;i<get_all_products().size();i++){
+
+    public List<WebElement> get_all_products() {
+        return get_list_of_elements(allProductElements);
+    }
+
+    public List<String> getProductsName() {
+        List<String> allProducts = new ArrayList<>();
+        for (int i = 0; i < get_all_products().size(); i++) {
             allProducts.add(get_all_products().get(i).getText());
         }
         return allProducts;
-}
-public void addToCart(int position){
+    }
 
-performMouseHover(get_all_products().get(position));
-clickUsingJavaScript(get_list_of_elements(AddToCartButtons).get(position));
-}
-public void continueShopping(){
-        clickUsingJavaScript(WaitElementToBeLocated(continueShopping));
-}
-public CartPage viewCart(){
+    public void scrollToSpecificProduct(int pixel) {
+        scrollVertically(pixel);
+    }
 
-    clickUsingJavaScript(WaitElementToBeLocated(viewCartButton));
-    return new CartPage(driver);
-}
+    public void addToCart(int position) {
+        performMouseHover(get_all_products().get(position));
+        clickUsingJavaScript(get_list_of_elements(AddToCartButtons).get(position));
+    }
+
+    public void continueShopping() {
+        clickUsingJavaScript(waitElementToBeLocated(continueShopping));
+    }
+
+    public CartPage viewCart() {
+
+        clickUsingJavaScript(waitElementToBeLocated(viewCartButton));
+        return new CartPage(driver);
+    }
+
+    public boolean allSearchResultIsVisible() {
+        for (int i = 0; i < get_all_products().size(); i++) {
+            if (get_all_products().get(i).isDisplayed() == false)
+                return false;
+        }
+        return true;
+    }
+
```

Comment:
I can not understand the point out of this method !!
I see this as a complex code that can be simplified